scripts/SolCx/pureshear_ALE_restart.jl

```diff
@@ -0,0 +1,215 @@
+        V.x[I...] =  verts.x[I[1]]*ε̇bg
+    end
+    if I[1]<=size(V.y,1) &&  I[2]<=size(V.y,2)
+        V.y[I...] = -verts.y[I[2]]*ε̇bg
+    end
+return nothing
+end
+
+
+function main()
+
+    @printf("Running on %d thread(s)\n", nthreads())
+
+    # Load Breakpoint data
+    if restart
+        file  = @sprintf("./Breakpoint%05d.jld2", last_step)
+        data  = load(file)
+        particles     = data["particles"]
+        phases        = data["phases"]   
+        phase_ratios  = data["phase_ratios"]
+        particle_args = data["particle_args"]
+        xlims = data["xlims"]
+        ylims = data["ylims"]
+        t     = data["t"]
+        Nt    = last_step + 100
+        it0   = last_step + 1
+        ε̇bg   = -1.
+    else
+        xlims = [-0.5 , 0.5]
+        ylims = [-0.5 , 0.5]
+        t     = 0.
+        Nt    = 100
+        it0   = 1
+        ε̇bg   = 1.
+    end
+
+    # Parameters
+    L    = (x =(xlims[2]-xlims[1]), y =(ylims[2]-ylims[1]) )
+    Nc   = (x=41, y=41 )
+    Nv   = (x=Nc.x+1,   y=Nc.y+1   )
+    Δ    = (x=L.x/Nc.x, y=L.y/Nc.y )
+    Nout = 10
+    C    = 0.25
+
+    # Model extent
+    cents_ext = (
+        x      = LinRange(xlims[1]-Δ.x/2, xlims[2]+Δ.x/2, Nc.x+2),
+        y      = LinRange(ylims[1]-Δ.y/2, ylims[2]+Δ.y/2, Nc.y+2),
+    )
+    cents     = (
+        x      = LinRange(xlims[1]+Δ.x/2, xlims[2]-Δ.x/2, Nc.x+2),
+        y      = LinRange(ylims[1]+Δ.y/2, ylims[2]-Δ.y/2, Nc.y+2),
+    )
+    verts = (
+        x      = LinRange(xlims[1], xlims[2], Nc.x+1),
+        y      = LinRange(ylims[1], ylims[2], Nc.y+1),
+    )
+    xlims     = [verts.x[1] , verts.x[end]]
+    ylims     = [verts.y[1] , verts.y[end]]
+
+    # Arrays
+    size_x    = (Nc.x+1, Nc.y+2)
+    size_y    = (Nc.x+2, Nc.y+1)
+    V = (
+        x = @zeros(size_x),
+        y = @zeros(size_y),
+    )
+
+    # Set velocity field
+    @parallel SetVelocity(V, verts, ε̇bg)
+ 
+    if !restart 
+        # Initialize particles -------------------------------
+        nxcell, max_xcell, min_xcell = 12, 24, 5
+        particles = init_particles(
+            backend, 
+            nxcell, 
+            max_xcell,
+            min_xcell, 
+            values(verts),
+            values(Δ),
+            values(Nc)
+        ) # random position by default
+
+        # Initialise phase field
+        particle_args = phases, = init_cell_arrays(particles, Val(1))  # cool
+
+        @parallel InitialFieldsParticles!(phases, particles.coords..., particles.index)
+    end
+
+    phase_ratios = JustPIC._2D.PhaseRatios(backend, 2, values(Nc));
+    phase_ratios_vertex!(phase_ratios, particles, values(verts), phases) 
+    phase_ratios_center!(phase_ratios, particles, values(verts), phases) 
+    Npart = sum(particles.index.data)  
+
+    println(" 
+    it => 0
+        extrema phase ratio @ vertices = $(extrema(sum(phase_ratios.vertex.data, dims=2)))
+        extrema phase ratio @ centers = $(extrema(sum(phase_ratios.center.data, dims=2)))
+    ")
```

Comment:
```suggestion
```

I guess we can remove this, it was mainly for debugging purposes

Reply:
OK